dev/Dockerfile

```diff
@@ -9,7 +9,9 @@ RUN apt-get install -y --no-install-recommends \
         libsasl2-modules
 ENV SETUPTOOLS_USE_DISTUTILS=stdlib
 
-COPY ../  ${AIRFLOW_HOME}/astro_sdk/
+COPY ../src  ${AIRFLOW_HOME}/astro_sdk/
```

Comment:
This will still mean that you need to rebuild the entire image when we make any changes in source code as all the source code is in `src`

Reply:
Check how we do it in astronomer-providers - https://github.com/astronomer/astronomer-providers/blob/main/dev/Dockerfile

Reply:
https://github.com/astronomer/astro-sdk/pull/676/commits/af32a9da73a92a26f34e5ff8966f4d8e8537fdce should fix it :)

Reply:
With `.dockerignore` you can revert the change to the `COPY` command no? `COPY ../` is fine if we keep our `.dockerignore` up to date as we do with `.gitignore`.

Reply:
right, we can do that.
Also, I can see we a `.dockerignore` at the root level so it was ignoring the newly added `dev/.dockerignore`. I have deleted the   `dev/.dockerignore` and updated  `.dockerignore` 3cb6012b76ee69742921d1318a28da2b7d864ef2

Reply:
The problem is the "src" folder that contains the entire source code, if we don't add it to .gitignore or specific files in Dockerfile, everytime I change something in `src/astro/database` for example, it will invalidate the cache and build it from scratch @pankajastro

Reply:
the following line is the problem: 

https://github.com/astronomer/astro-sdk/blob/3cb6012b76ee69742921d1318a28da2b7d864ef2/dev/Dockerfile#L12

We only want to install the dependencies in the docker-file, and then mount the sources via docker-compose file

Reply:
Validate this by changing a comment in https://github.com/astronomer/astro-sdk/blob/3cb6012b76ee69742921d1318a28da2b7d864ef2/src/astro/files/__init__.py and run `make container target=build-run` command again

Reply:
Yes, I agree @kaxil. We can either add `src` also in the `.dockerignore` or we will have to copy only `README.md` and  `__init__.py` which are required by `pyproject.toml` I think the 2nd option is more clean and easy so I have reverted the `Dockerfile` changes here 736149e623f5fb7126cb85e4088b9f32d3dece5a